topdown/strings.go

```diff
@@ -102,10 +103,16 @@ func builtinIndexOf(a, b ast.Value) (ast.Value, error) {
 	}
 
 	baseRunes := []rune(string(base))
-	searchRune := []rune(string(search))[0]
+	searchRunes := []rune(string(search))
+	searchLen := len(searchRunes)
+
 	for i, r := range baseRunes {
-		if r == searchRune {
-			return ast.IntNumberTerm(i).Value, nil
+		if r == searchRunes[0] {
+			if len(baseRunes) >= i+searchLen {
+				if reflect.DeepEqual(baseRunes[i:i+searchLen], searchRunes) {
```

Comment:
Huh. Using reflect here comes as a surprise. If we can use a for loop instead, I'd prefer that... `reflect` is always a bit of a warning sign wrt performance, I suppose we could also do some benchmarking here... 🤔